### ISSUE TITLE
Override queue options [alpha]

### ISSUE THREAD

**Issue by ruanitto** [Rocketseat/adonis-bull#64]
**Description of bug**

As described in the alpha version documentation, it is possible to overwrite the queue / job options, however the options are not applied at the time of execution.
As the example below, I configured some options, among them the removeOnComplete option, when adding the Job in a queue the options are not inserted.
In the adonis4 version these settings work correctly 

**To Reproduce**
Configure Job:

```JS
export default class AlertsJob implements JobContract {
  public key = 'alertsJob'

  public options: JobsOptions = {
    delay: 1500,
    attempts: 2,
    removeOnComplete: true,
    backoff: {
      type: 'exponential',
      delay: 3000,
    },
  }
....
```

**Expected behavior**

After the Job runs without errors, it must be removed from the queue 

**Exception or Error**
Job remains in queue and the options defined are not part of the data

<pre><code>

</code></pre>

**Screenshots**

![image](https://user-images.githubusercontent.com/13773521/106212756-03bd3c00-61aa-11eb-8688-edc6e29fb32a.png)

**Environment:**

**Additional context**

**Comment by rodrigorz** [comment]
Also in the adonis4 version is possible to set `defaultJobsOptions`, i didn't see this option in the alpha version.

The `defaultJobsOptions` is good to set the same config for all jobs, like `removeOnComplete`, `backoff`, etc.

Maybe it can be done along with this issue.

**Comment by rodrigorz** [comment]
One more thing, using the example above:

Inside my Controller, if i try to (following the docs)
```js
Bull.add(AlertsJob.key, data);
```
i got this error message `Property 'key' does not exist on type 'typeof AlertsJob'.ts(2339)`.

But this way it works, creating a job object.
```js
Bull.add(new AlertsJob().key, data);
```

Is that so or did I do something wrong?

**Comment by ruanitto** [comment]
> One more thing, using the example above:
> 
> Inside my Controller, if i try to (following the docs)
> 
> ```js
> Bull.add(AlertsJob.key, data);
> ```
> 
> i got this error message `Property 'key' does not exist on type 'typeof AlertsJob'.ts(2339)`.
> 
> But this way it works, creating a job object.
> 
> ```js
> Bull.add(new AlertsJob().key, data);
> ```
> 
> Is that so or did I do something wrong?

In my case, since version 4 I have always preferred to use the job key directly in the method, without having to import the job into the controller:

**Comment by rodrigorz** [comment]
> In my case, since version 4 I have always preferred to use the Job key directly in the method, without having to import the Job into the controller:
> 
> ```js
> Bull.add('alertsJob', data);
> ```

In version 4 the `AlertsJob.key` works, because of the method `static get key() {}`.

I like to import the Job, it is easy to guarantee the use of the correct name and if at any time i need to rename the job name, it is simple.

But the `new AlertsJob().key` way works fine, so this is just a mention of something that is different in the docs. 😄

**Comment by rodrigorz** [comment]
I just solve the "problem" with this
```js
export default class AlertsJob implements JobContract {
  public key = 'AlertsJob';

  public static get key() {
    return this.key;
  }

  // ..... options and handle
}
```

**Comment by ruanitto** [comment]
I could also declare the key as static:

```js
export default class AlertsJob implements JobContract {
  static public key = 'AlertsJob';
//...
}
``

**Comment by rodrigorz** [comment]
> You can also declare the key as static:
> 
> ```js
> export default class AlertsJob implements JobContract {
>   public static key = 'AlertsJob';
> //...
> }
> ```

The `JobContract` interface does not allow this. The 'key' can't be static.

```js
Class 'AlertsJob' incorrectly implements interface 'JobContract<any>'.
  Property 'key' is missing in type 'AlertsJob' but required in type 'JobContract<any>'.ts(2420)
```

**Comment by HigoRibeiro** [comment]
Hi @ruanitto, thanks a lot to report this issue!

I checked and actually can't get `options`.
I'll create a PR to fix this.

**Comment by HigoRibeiro** [comment]
> Also in the adonis4 version is possible to set `defaultJobsOptions`, i didn't see this option in the alpha version.
> 
> The `defaultJobsOptions` is good to set the same config for all jobs, like `removeOnComplete`, `backoff`, etc.
> 
> Maybe it can be done along with this issue.
@rodrigorz:

The `defaultJobsOptions` now is called just `options`, and I'll fix this.

I'll see about your `key` point as well

**Comment by rodrigorz** [comment]
> > Also in the adonis4 version is possible to set `defaultJobsOptions`, i didn't see this option in the alpha version.
> > The `defaultJobsOptions` is good to set the same config for all jobs, like `removeOnComplete`, `backoff`, etc.
> > Maybe it can be done along with this issue.
> > @rodrigorz:
> 
> The `defaultJobsOptions` now is called just `options`, and I'll fix this.
> 
> I'll see about your `key` point as well

Thanks.

The work around that i mentioned before won't work, this will cause an infinity loop. 
```js
  public static get key(): string {
    return this.key;
  }
```

The `public static get` method must have a different name from the `key` attribute.

**Comment by rodrigorz** [comment]
> > Also in the adonis4 version is possible to set `defaultJobsOptions`, i didn't see this option in the alpha version.
> > The `defaultJobsOptions` is good to set the same config for all jobs, like `removeOnComplete`, `backoff`, etc.
> > Maybe it can be done along with this issue.
> > @rodrigorz:
> 
> The `defaultJobsOptions` now is called just `options`, and I'll fix this.
> 
> I'll see about your `key` point as well

Thanks.

By the way, the work around that i mentioned before won't work, it will cause an infinity loop.
```js
public static get key(): string {
  return this.key; // this return will execute the method again =/
}
```

Perhaps changing the name of the method will work. But setting the `key` attribute to static seems to be the way to go.

**Comment by jpedroschmitz** [comment]
Hey, I have just released `v1.0.0-2` that should fix this issue!

To install it, just run `yarn add @rocketseat/adonis-bull@alpha@latest`.

Thanks for your patience and help in reporting the bug!
